collated/main/A0144061U.md

```diff
@@ -0,0 +1,606 @@
+``` java
+/**
+ * Undo the most recent command by the user.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+    		+ ": Undo the latest command. If there is no previous command, nothing will happen.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_UNDO_SUCCESS = "Undone the latest task.";
+    public static final String MESSAGE_UNDO_FAIL = "There was no undoable command made before.";
+
+    public UndoCommand() {}
+
+    @Override
+    public CommandResult execute() {
+        try {
+            model.undoLatestCommand();
+        } catch (EmptyStackException e) {
+        	return new CommandResult(MESSAGE_UNDO_FAIL);
+        }
+
+        return new CommandResult(MESSAGE_UNDO_SUCCESS);
+    }
+
+}
+```
+###### /java/seedu/todoList/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getFilteredTodoList() {
+        return model.getFilteredTodoList();
+    }
+    
+```
+###### /java/seedu/todoList/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getFilteredEventList() {
+        return model.getFilteredEventList();
+    }
+    
+```
+###### /java/seedu/todoList/logic/LogicManager.java
+``` java
+    @Override
+    public ObservableList<ReadOnlyTask> getFilteredDeadlineList() {
+        return model.getFilteredDeadlineList();
+    }
+}
+```
+###### /java/seedu/todoList/MainApp.java
+``` java
+    /*
+     * Stop and restart the app with new storage file
+     */
+    @Subscribe
+    private void handleStorageLocationChangedEvent (StorageLocationChangedEvent event) throws Exception {
+    	logger.info(LogsCenter.getEventHandlingLogMessage(event));
+    	
+    	// ui.stop();
+    	
+    	logger.info("=============================[ Reinitializing TodoList ]===========================");
+        super.init();
+        
+        String newDirectory = event.getNewDirectory();
+        Config changedConfig = new Config(newDirectory);
+        ConfigUtil.saveConfig(changedConfig, Config.DEFAULT_CONFIG_FILE);
+
+        this.stop();
+    }
+
+    public static void main(String[] args) {
+        launch(args);
+    }
+```
+###### /java/seedu/todoList/model/task/UniqueTaskList.java
+``` java
+    /**
+     * Removes all task from the list.
+     */
+    public void removeAll() {
+       internalList.remove(0, internalList.size());
+    }
+    
+    /**
+     * Replace all task from another list.
+     */
+    public void setAll(UniqueTaskList taskList) {
+       internalList.setAll(taskList.getInternalList());
+    }
+```
+###### /java/seedu/todoList/model/TaskList.java
+``` java
+    public void resetData(Collection<? extends ReadOnlyTask> newTasks) {
+    	Object[] typeCheck = newTasks.toArray();
+    	if(typeCheck.length == 0) {
+    		tasks = new UniqueTaskList();
+    	}
+    	else if(typeCheck[0] instanceof Todo) {
```

Comment:
coding standard violation, `else` / `else if` should be placed between the braces